tests/canary_upgrade_test.go

```diff
@@ -0,0 +1,195 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/flags"
+	"kubevirt.io/kubevirt/tests/util"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("[Serial][sig-compute]virt-handler canary upgrade", func() {
+
+	var err error
+	var originalKV *v1.KubeVirt
+	var virtCli kubecli.KubevirtClient
+	var dsInformer cache.SharedIndexInformer
+	var stopCh chan struct{}
+	var lastObservedEvent string
+
+	const e2eCanaryTestAnnotation = "e2e-canary-test"
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+
+		virtCli, err = kubecli.GetKubevirtClient()
+		Expect(err).ToNot(HaveOccurred())
+
+		originalKV = util.GetCurrentKv(virtCli).DeepCopy()
+
+		stopCh = make(chan struct{})
+		lastObservedEvent = ""
+
+		informerFactory := informers.NewSharedInformerFactoryWithOptions(virtCli, 0, informers.WithNamespace(flags.KubeVirtInstallNamespace), informers.WithTweakListOptions(func(opts *metav1.ListOptions) {
+			opts.LabelSelector = "kubevirt.io=virt-handler"
+		}))
+		dsInformer = informerFactory.Apps().V1().DaemonSets().Informer()
+	})
+
+	AfterEach(func() {
```

Comment:
Don't we need to wait for handlers to be ready again?

Reply:
You're right we probably should.

Reply:
Fixed it!

Reply:
@xpivarc can you unhold now that this is fixed?